### PR TITLE
Enable yapf (Python formatting)

### PR DESCRIPTION
**Pull request by soltanmm-google**
supercedes #6544

### CODE REVIEW

File: tools/distrib/pyformat_code.sh [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+set -ex
+
+# change to root directory
+cd $(dirname $0)/../..
+
+DIRS=src/python
+EXCLUSIONS='src/python/grpcio/grpc_*.py src/python/grpcio_health_checking/grpc_*.py src/python/grpcio_reflection/grpc_*.py src/python/grpcio_tests/grpc_*.py'
```
Comment (kpayson64) [inline tools/distrib/pyformat_code.sh]: What do you think about modifying the auto-generators for these files to create compliant code so we don't need exclusions?

File: tools/distrib/pyformat_code.sh [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+exclusion_args=""
+for exclusion in $EXCLUSIONS; do
+  exclusion_args="$exclusion_args --exclude $exclusion"
+done
+
+script_result=0
+for dir in $DIRS; do
+  tempdir=`mktemp -d`
+  cp -RT $dir $tempdir
+  $PYTHON -m pyformat -i -r -j0 $exclusion_args $dir
```
Comment (kpayson64) [inline tools/distrib/pyformat_code.sh]: To avoid the copy, can we just do non- inplace pyformat?

Comment (soltanmm-google) [inline tools/distrib/pyformat_code.sh]: I think the point of that was to allow one call to both format the code and check for deltas, so there wouldn't be a second command file cluttering up the distrib folder.

File: tools/distrib/pyformat_code.sh (hunk 1 above)
Comment (nathanielmanistaatgoogle) [inline tools/distrib/pyformat_code.sh]: I support it... lightly. I wouldn't hold up this change insisting on it.

Comment (nathanielmanistaatgoogle) [inline tools/distrib/pyformat_code.sh]: ... wait a second, what files are captured here and why do they have to be excluded?

**Comment by nathanielmanistaatgoogle** [comment]
Woohoo!

Is there anything like a `.pyformatrc` that we should keep around or are we confident that default-settings `pyformat` is what we want?

**Comment by ctiller** [comment]
It's probably worth generating something (even that says 'use the
defaults'): it's helped with clang-format to have a place where debates can
happen in PR's.

File: tools/distrib/pyformat_code.sh (hunk 2 above)
Comment (kpayson64) [inline tools/distrib/pyformat_code.sh]: If pyformat doesn't have a diff tool built in, this seems reasonable then

**Comment by soltanmm-google** [comment]
The question is whether or not `pyformat` supports settings at all. It uses a tool that *might* support settings, so maybe indirectly setting the settings for the underlying tool will work (... hacky...), but that needs a bit of digging.

**Comment by soltanmm-google** [comment]
Configuration is available under `$REPO_ROOT/setup.cfg`'s `[yapf]` section.

File: tools/distrib/pyformat_code.sh (hunk 1 above)
Comment (soltanmm-google) [inline tools/distrib/pyformat_code.sh]: All the auto-generated files from the build templates.

**Pull request merged**